### PR TITLE
Address failing HelixWatch3DViewModelTests in the WpfVisualizationTests

### PR DESCRIPTION
**Pull request by alfarok**
### Purpose

[DYN-1572](https://jira.autodesk.com/browse/DYN-1572)

This PR addresses the failing `HelixWatch3DViewModelTests` in the `WpfVisualizationTests`

These tests seems to be failing due to timing issues between opening a graph and the UI being updated.  By subscribing to `EvaluationCompleted` on the `DynamoModel` we can call `DispatcherUtil.DoEvents` immediately following to force the dispatcher to dump the entire queue.  `NUnit` has some special behavior regarding `async void` methods and how they interact with the `SynchronizationContext`, more info on this can be found [here](https://stackoverflow.com/questions/15031681/how-does-nunit-successfully-wait-for-async-void-methods-to-complete/15032244#15032244).

image

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner @aparajit-pratap 

### FYIs

@reddyashish

### CODE REVIEW

File: src/VisualizationTests/HelixWatch3DViewModelTests.cs [hunk 1]
```diff
@@ -97,6 +97,8 @@ protected override void StartDynamo(TestSessionConfiguration testConfig)
                     ProcessMode = TaskProcessMode.Synchronous
                 });
 
+            Model.EvaluationCompleted += Model_EvaluationCompleted;
```
Comment (mjkkirschner) [inline src/VisualizationTests/HelixWatch3DViewModelTests.cs]: just to be sure, unsubscribe this handler when the test teardown.

**Comment by mjkkirschner** [comment]
@alfarok the explanation in that link sounds kind of like the explanation for doing `SetTimeout(()=>{},0)` in js - does that sound right?

**Comment by alfarok** [comment]
@mjkkirschner yes very similar I think, something along the lines of pausing the execution to let the rendering threads catch up

**Comment by mjkkirschner** [comment]
LGTM - but are you also going to modify the test jobs to include these dlls?

**Comment by alfarok** [comment]
@mjkkirschner yes we need to do that but @reddyashish is still wrapping up the remaining 3 custom node failures which are part of the `WpfVisualizationTests`

**Pull request merged**